src/index.js

```diff
@@ -68,11 +70,11 @@ const decorator = ({ declareQueries, declareCommands, declareConnect }) => (Comp
   const queriesInputTypes = queries && Object.keys(declaredQueries.InputType) || [];
   const declaredCommands = commands && declareCommands(commands);
   const commandsInputTypes = commands && Object.keys(declaredCommands.InputType) || [];
-  const declaredConnect = (connect || local || queries || commands) && declareConnect([
+  const declaredConnect = (connect || local || queries || commands) && declareConnect(intersection([
     ...difference(queriesInputTypes, Object.keys(local || {})),
     ...difference(commandsInputTypes, Object.keys(local || {})),
     ...(connect || [])
-  ].concat(local ? ['___local'] : []));
+  ].concat(local ? ['___local'] : []), Object.keys(appState.meta.props)));
```

Comment:
I think we should do this `intersection` only on `queriesInputTypes` and `commandsInputTypes`, otherwise a user could write, say, `connect: ['nonExistingKey']` and wouldn't get any warning